happi/cli.py

```diff
@@ -1075,6 +1077,73 @@ def audit(
         click.echo(f'# devices failed: {len(unique_fails)} / {len(results)}')
 
 
+@happi_cli.command()
+@click.pass_context
+@click.option('--fix-optional/--ignore-optinal', 'fix_optional', default=False,
+              help='Also prompt for user input on optional information')
+@click.option('--glob/--regex', 'use_glob', default=True,
+              help='Use glob (default) or regex style search terms. '
+              'Only relevant if search_criteria are provided.')
+@click.argument('search_criteria', nargs=-1)
+def repair(
+    ctx,
+    fix_optional: bool,
+    use_glob: bool,
+    search_criteria: tuple[str, ...]
+):
+    """
+    Repair the database.
+
+    Repairs all entries matching SEARCH_CRITERIA, repairs entire database otherwise.
+    """
+    logger.debug('starting repair block')
+
+    client: happi.Client = ctx.obj
+    if search_criteria:
+        results = search_parser(client, use_glob, search_criteria)
+    else:
+        # run repair on all items
+        results = search_parser(client, True, '*')
+
+    for res in results:
+        # fix mandatory info with missing defaults
+        req_info = (info for info in find_unfilled_mandatory_info(res))
+
+        if fix_optional:
+            empty_opt = find_unfilled_optional_info(res)
+            req_info = itertools.chain(req_info, empty_opt)
+
+        try:
+            req_field = next(req_info)
+        except StopIteration:
+            req_field = False
+
+        res_name = res['_id']
```

Comment:
Seeing `_id` reminded me of something tangential to "missing info" but I think likely something `happi repair` could (should?) handle.

I think we've discussed and agreed (correct me if I'm misremembering) that `name` and `_id` should be identical. This tool could check if that's the case and fix it, replacing `_id` with the value in `name`.

Follow-up PR, perhaps?

Reply:
I think a follow-up PR would be a good place to address this.  I envision this tool growing as we find different things to fix.